dockerfiles/packaging/guice/cassandra/scripts/james.service

```diff
@@ -1,8 +1,8 @@
 [Unit]
 Description=James stands for Java Apache Mail Enterprise Server! It has a modular architecture based on a rich set of modern and efficient components which provides at the end complete, stable, secure and extendable Mail Servers running on the JVM.
 Documentation=http://james.apache.org
-Wants=network-online.target cassandra.service elasticsearch.service
-After=network-online.target cassandra.service elasticsearch.service
+Wants=network-online.target cassandra.service elasticsearch.service rabbitmq.service
```

Comment:
it seems you are doing it the right way! Wants indicates that the dependency is optional, while after ensure the dependency order. So if RabbitMQ is not installed, the service will run without it, while if it is it will start RabbitMQ first.